apps/state/lib/state/schedule.ex

```diff
@@ -141,8 +141,25 @@ defmodule State.Schedule do
     {:ok, overridden_state, timeout_or_hibernate}
   end
 
-  # Converts routes and stops into workable ids
   @spec convert_filters(filter_opts) :: convert_filters
+  defp convert_filters(%{routes: routes, stops: stops} = filters) do
+    route_ids =
+      routes
+      |> Stream.uniq()
+      |> MapSet.new()
+
+    stop_ids =
+      stops
+      |> Stream.uniq()
+      |> Enum.filter(&stop_on_routes?(&1, route_ids))
+
+    filters
+    |> Map.delete(:routes)
+    |> Map.put(:stops, stop_ids)
+    |> convert_filters()
```

Comment:
You're right in that this doesn't quite work. Bus stops can have multiple routes which stop there, so this doesn't catch that case. For example, try http://localhost:4000/schedules?filter[route]=9&filter[stop]=44 where you'll also get schedules for the 5 and 10.